src/asherah.ts

```diff
@@ -149,3 +151,30 @@ export function encrypt(partitionId: string, data: Buffer): DataRowRecord {
 
     return dataRowRecord;
 }
+
+export function decrypt_from_json_string(partitionId: string, dataRowRecord: string): Buffer {
+  const partitionIdBuffer = string_to_cbuffer(partitionId);
+  const jsonBuffer = string_to_cbuffer(dataRowRecord);
+  const outputDataBuffer = allocate_cbuffer(jsonBuffer.byteLength);
```

Comment:
Does this need to be the full length of the JSON, or just the length of the "data" element? That would introduce the non-ideal need to deserialize the JSON string, but I'm not sure which is worse -- allocating too much, or JSON deserialization

Reply:
Allocating too much is vastly faster